sky/clouds/azure.py

```diff
@@ -295,6 +295,20 @@ def make_deploy_resources_variables(
                 content: |
                   APT::Periodic::Enable "0";
             """).encode('utf-8')).decode('utf-8')
+
+        def _failover_disk_tier() -> Optional[resources_utils.DiskTier]:
+            # Failover disk tier.
+            all_tiers = list(reversed(resources_utils.DiskTier))
+            start_index = all_tiers.index(
+                Azure._translate_disk_tier(r.disk_tier))
+            while start_index < len(all_tiers):
+                disk_tier = all_tiers[start_index]
+                ok, _ = Azure.check_disk_tier(r.instance_type, disk_tier)
+                if ok:
+                    return disk_tier
+                start_index += 1
+            return None
```

Comment:
Does this mean a user specifying `--disk-tier medium` may get an instance with disk_tier low?

Reply:
Good point! We should only do failover if the disk tier is None or BEST. Just pushed a commit to fix that.

Reply:
Just re-run every tests and passed